src/avram/needy_initializer_and_save_methods.cr

```diff
@@ -165,12 +164,8 @@ module Avram::NeedyInitializerAndSaveMethods
       {% if with_bang %}
         operation.update!
       {% else %}
-        if operation.save
-          yield operation, operation.record.not_nil!
-        else
-          operation.published_save_failed_event
```

Comment:
This was moved to `SaveOperation#save` so we don't need to remember to call it in multiple places